dev/src/lobster/typecheck.h

```diff
@@ -219,6 +219,22 @@ struct TypeChecker {
         }
     }
 
+    bool ConvertsToStatically(TypeRef type, TypeRef bound) {
+        if (bound == type || type->t == V_ANY)
+            return true;
+        if (bound->t != V_CLASS || type->t != V_CLASS)
+            return false;
+        if (st.SuperDistance(type->udt, bound->udt) > -1)
+            return true;
+        if (!TypeOfUndefinedGenericSpec(type))
+            return false;
+        for (auto t = bound->udt; t; t = t->resolved_superclass) {
```

Comment:
this checks not by udt itself, but by `first` reference, because at this point only base type is interesting, not actual specializations. I'm not sure if/how could I make this work with SuperDistance..

Reply:
as per #99 this will be replaced by checking first just on bound without looping over parents